sup3r/preprocessing/samplers/cc.py

```diff
@@ -0,0 +1,179 @@
+            batch_size=batch_size,
+            t_enhance=t_enhance,
+            s_enhance=s_enhance,
+            feature_sets=feature_sets,
+        )
+
+    def check_for_consistent_shapes(self):
+        """Make sure container shapes are compatible with enhancement
+        factors."""
+        enhanced_shape = (
+            self.lr_data.shape[0] * self.s_enhance,
+            self.lr_data.shape[1] * self.s_enhance,
+            self.lr_data.shape[2] * (1 if self.t_enhance == 1 else 24),
+        )
+        msg = (
+            f'hr_data.shape {self.hr_data.shape} and enhanced '
+            f'lr_data.shape {enhanced_shape} are not compatible with '
+            f'the given enhancement factors t_enhance = {self.t_enhance}, '
+            f's_enhance = {self.s_enhance}'
+        )
+        assert self.hr_data.shape[:3] == enhanced_shape, msg
+
+    def reduce_high_res_sub_daily(self, high_res, csr_ind=0):
+        """Take an hourly high-res observation and reduce the temporal axis
+        down to lr_sample_shape[2] * t_enhance time steps, using only daylight
+        hours on the middle part of the high res data.
+
+        Parameters
+        ----------
+        high_res : T_Array
+            5D array with dimensions (n_obs, spatial_1, spatial_2, temporal,
+            n_features) where temporal >= 24 (set by the data handler).
+        csr_ind : int
+            Feature index of clearsky_ratio. e.g. self.data[..., csr_ind] ->
+            cs_ratio
+
+        Returns
+        -------
+        high_res : T_Array
+            5D array with dimensions (n_obs, spatial_1, spatial_2, temporal,
+            n_features) where temporal has been reduced down to the integer
+            lr_sample_shape[2] * t_enhance. For example if hr_sample_shape[2]
+            is 9 and t_enhance = 8, 72 hourly time steps will be reduced to 9
+            using the center daylight 9 hours from the second day.
+
+        Note
+        ----
+        This only does something when `1 < t_enhance < 24.` If t_enhance = 24
+        there is no need for reduction since every daily time step will have 24
+        hourly time steps in the high_res batch data. Of course, if t_enhance =
+        1, we are running for a spatial only model so this routine is
+        unnecessary.
+
+        *Needs review from @grantbuster
+        """
+        if self.t_enhance not in (24, 1):
+            high_res = self.get_middle(high_res, self.hr_sample_shape)
+            high_res = nsrdb_reduce_daily_data(
+                high_res, self.hr_sample_shape[-1], csr_ind=csr_ind
+            )
+        return high_res
+
+    @staticmethod
+    def get_middle(high_res, sample_shape):
+        """Get middle chunk of high_res data that will then be reduced to day
+        time steps. This has n_time_steps = 24 if sample_shape[-1] <= 24
+        otherwise n_time_steps = sample_shape[-1]."""
+        n_days = int(high_res.shape[3] / 24)
+        if n_days > 1:
```

Comment:
What if `n_days == 1`? at least raise an explicit error maybe?

Reply:
Yeah I think it shouldnt make it here if the enhancement factor results in high_res.shape[3] == 24. I think that would require t_enhance = 24 and sample_shape[2] = 24. Could do a check on the given t_enhance value to account for this

Reply:
should probably be stopped somewhere, either in the init or in this function. Can you just add an assert statement?

Reply:
This can actually happen also if sample_shape[2] = 24 and t_enhance = 1. Which should be allowed right? Im confused why n_days < 1 and n_days > 1 but not n_days = 1 should be allowed here